cmake/flash/CMakeLists.txt

```diff
@@ -36,8 +36,17 @@ function(runners_yaml_append_config)
   get_runners_prop(elf_file elf "${KERNEL_ELF_NAME}")
   runners_yaml_append("  # Build outputs:")
   runners_yaml_append("  elf_file: ${elf}")
+
+  if(SYSBUILD)
+    zephyr_get(alt_file VAR ALT_SIGNED_HEX_FILE SYSBUILD)
+  endif()
+
   if(CONFIG_BUILD_OUTPUT_HEX)
-    get_runners_prop(hex_file hex "${KERNEL_HEX_NAME}")
+    if(SYSBUILD AND DEFINED alt_file)
+      set(hex "${alt_file}")
+    else()
```

Comment:
not directly related to those changes themselves, but I was wondering if this would be a use-case where `sysbuild` should be able to support flashing of specific binaries / hex files.

That way, sysbuild could mark the `app` image as build only, and instead specify its own alternate image as a custom hex file to flash.

Reply:
Could be a possible expansion, though there area things like https://github.com/zephyrproject-rtos/zephyr/pull/64551 which throw a spanner in the works

Reply:
> not directly related to those changes themselves, but I was wondering if this would be a use-case where `sysbuild` should be able to support flashing of specific binaries / hex files.

it is not support well for `west flash`, as it heavyily depends on  CMakeCache.txt and runners.yaml, which is generated by west build